dataset/src/main/kotlin/com/cosmotech/dataset/service/DatasetServiceImpl.kt

```diff
@@ -831,9 +815,16 @@ class DatasetServiceImpl(
       workspaceId: String
   ): Dataset {
     val dataset = findDatasetById(organizationId, datasetId)
-    dataset.linkedWorkspaceIdList?.add(workspaceId)
-        ?: run { dataset.linkedWorkspaceIdList = mutableListOf(workspaceId) }
-
+    val linkedWorkspaceIdList = dataset.linkedWorkspaceIdList
+    if (linkedWorkspaceIdList != null) {
+      if (linkedWorkspaceIdList.contains(workspaceId)) {
+        return dataset
+      } else {
+        dataset.linkedWorkspaceIdList!!.add(workspaceId)
+      }
+    } else {
+      dataset.linkedWorkspaceIdList = mutableListOf(workspaceId)
+    }
```

Comment:
Minor suggestion, but readability could probably be improved with something like this:
```
if (dataset.linkedWorkspaceIdList == null) {
  dataset.linkedWorkspaceIdList = mutableListOf(workspaceId)
} else if (dataset.linkedWorkspaceIdList.contains(workspaceId)) {
  return dataset
} else {
  dataset.linkedWorkspaceIdList!!.add(workspaceId)
}
```
(code should be double checked though, I don't know kotlin :stuck_out_tongue: )

Same thing for `addDatasetToLinkedDatasetIdList` in the workspace service

Reply:
Done